### ISSUE TITLE
Can't parse simpe TensorFlow model. Error:Error parsing text-format opencv_tensorflow.GraphDef: 714:5: Unknown enumeration value of "DT_RESOURCE" for field "type"

### ISSUE THREAD

**Issue by bronstein87** [opencv/opencv#17962]
- OpenCV => 4.3
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2017

##### Detailed description

Hi, i'm trying to run some simple model, but it gives me an error during parse:
`Error parsing text-format opencv_tensorflow.GraphDef: 714:5: Unknown enumeration value of "DT_RESOURCE" for field "type".`
Is there any way to fix this issue?
It is cnn from deep sort alghoritm.
https://drive.google.com/file/d/11VKR4sMX2XIRov_qT9k4r7XDSnlBnNT9/view?usp=sharing - here are .pb and .pbtxt file

https://github.com/nwojke/deep_sort/blob/d303103724d38ee4b4734974986e1f1c9b4dc6ef/tools/freeze_model.py - here is a code to generate this model from scratch

**Comment by dkurt** [comment]
Have you tried without `.pbtxt`? It's optional

**Comment by bronstein87** [comment]
> Have you tried without `.pbtxt`? It's optional

Unfoturnatenly, I get another error in this case 
cv2.error: OpenCV(4.3.0) D:\soft\opencv\opencv-4.3.0\modules\dnn\src\tensorflow\tf_importer.cpp:545: error: (-2:Unspecified error) Input layer not found: Const in function 'cv::dnn::dnn4_v20200310::`anonymous-namespace'::TFImporter::connect'

**Comment by dkurt** [comment]
> OpenCV(4.3.0) 

Can you try the latest release?

**Comment by bronstein87** [comment]
> > OpenCV(4.3.0)
> 
> Can you try the latest release?

I tried 4.4.0 and still get same errors (with and without pbtxt) :(

With:
[libprotobuf ERROR D:\soft\opencv\opencv-4.4.0\3rdparty\protobuf\src\google\protobuf\text_format.cc:292] Error parsing text-format opencv_tensorflow.GraphDef: 1103:5: Unknown enumeration value of "DT_RESOURCE" for field "type".
cv2.error: OpenCV(4.4.0) D:\soft\opencv\opencv-4.4.0\modules\dnn\src\tensorflow\tf_io.cpp:54: error: (-2:Unspecified error) FAILED: ReadProtoFromTextFile(param_file, param). Failed to parse GraphDef file: D:/projects/deep_sort-master/RUNNINGDATA/tensor_networks/model_last.pbtxt in function 'cv::dnn::ReadTFNetParamsFromTextFileOrDie'
Without:

[libprotobuf ERROR D:\soft\opencv\opencv-4.4.0\3rdparty\protobuf\src\google\protobuf\text_format.cc:292] Error parsing text-format opencv_tensorflow.GraphDef: 1103:5: Unknown enumeration value of "DT_RESOURCE" for field "type".

**Comment by bronstein87** [comment]
> > OpenCV(4.3.0)
> 
> Can you try the latest release?

I also made an attempt to parse model from onnx, but again with no success:

cv2.error: OpenCV(4.4.0) D:\soft\opencv\opencv-4.4.0\modules\dnn\src\layers\convolution_layer.cpp:94: error: (-213:The function/feature is not implemented) Unsupported asymmetric padding in convolution layer in function 'cv::dnn::BaseConvolutionLayerImpl::BaseConvolutionLayerImpl'

**Comment by pedro-patlan** [comment]
Hi @bronstein87, did you have any luck? I'm facing the same issue and I was wondering if you find any solution.

Thanks.

**Comment by ebrarsahin** [comment]
@pedro-patlan Did you find any solution ?

**Comment by Tandon-A** [comment]
@bronstein87 Did you find any solution for this?

**Comment by bronstein87** [comment]
> @bronstein87 Did you find any solution for this?

Hi. No, I recommend you to use TensorRT for inference.

**Comment by Tandon-A** [comment]
> > @bronstein87 Did you find any solution for this?
> 
> Hi. No, I recommend you to use TensorRT for inference. Or you can use this library https://github.com/serizba/cppflow

Thanks for the advice. Will look into these options. 

For now, I was able to convert the model to ONNX format and load it through OpenCV. The source model though wasn't the TensorFlow one developed by N. Wojke, but the PyTorch one shared in [this repository](https://github.com/ZQPei/deep_sort_pytorch).
